### ISSUE TITLE
Deprication of now/0 and time precision issue in windows

### ISSUE THREAD

**Issue by c-bik** [K2InformaticsGmbH/imem#45]
In windows erlang uses [GetSystemTime](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724390%28v=vs.85%29.aspx) to get current system time:

``` erlang
> erlang:system_info(os_system_time_source).                                                                          
[{function,'GetSystemTime'},
 {resolution,100},
 {parallel,yes},
 {time,1482414054673408}]
```

This has the [following limitation](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724408%28v=vs.85%29.aspx)
&nbsp;&nbsp;&nbsp;**Remarks**
&nbsp;&nbsp;&nbsp;The resolution of the GetTickCount function is limited to the resolution of the system timer,
&nbsp;&nbsp;&nbsp;which is typically in the range of 10 milliseconds to 16 milliseconds. The resolution of the
&nbsp;&nbsp;&nbsp;GetTickCount function is not affected by adjustments made by the GetSystemTimeAdjustment
&nbsp;&nbsp;&nbsp;function.

As an alternative, [GetSystemTimePreciseAsFileTime](https://msdn.microsoft.com/en-us/library/windows/desktop/hh706895%28v=vs.85%29.aspx) can be used for Windows (NIF), which gives:
&nbsp;&nbsp;&nbsp;The GetSystemTimePreciseAsFileTime function retrieves the current
&nbsp;&nbsp;&nbsp;system date and time with the highest possible level of precision (<1us).
&nbsp;&nbsp;&nbsp;The retrieved information is in Coordinated Universal Time (UTC) format.

Current Resolution : 10 millisecond

**Comment by acautin** [comment]
in OS X I got:
```
erlang:system_info(os_system_time_source).
[{function,gettimeofday},
 {resolution,1000000},
 {parallel,yes},
 {time,1447671428706194000}]
```

**Comment by c-bik** [comment]
In CentOS release 6.6 (Final), 2.6.32-504.1.3.el6.x86_64 #1 SMP I got:
```erlang
Erlang/OTP 18 [erts-7.1] [source-2882b0c] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V7.1  (abort with ^G)
1> erlang:system_info(os_system_time_source).
[{function,clock_gettime},
 {clock_id,'CLOCK_REALTIME'},
 {resolution,1000000000},
 {parallel,yes},
 {time,1447672387678884590}]
```

**Comment by c-bik** [comment]
@acautin MAC To be same as Linux in clock resolution.
From https://gist.github.com/jbenet/1087739
```C++
/* 
author: jbenet
os x, compile with: gcc -o testo test.c 
linux, compile with: gcc -o testo test.c -lrt
*/

#include <time.h>
#include <sys/time.h>
#include <stdio.h>

#ifdef __MACH__
#include <mach/clock.h>
#include <mach/mach.h>
#endif


void current_utc_time(struct timespec *ts) {

#ifdef __MACH__ // OS X does not have clock_gettime, use clock_get_time
  clock_serv_t cclock;
  mach_timespec_t mts;
  host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
  clock_get_time(cclock, &mts);
  mach_port_deallocate(mach_task_self(), cclock);
  ts->tv_sec = mts.tv_sec;
  ts->tv_nsec = mts.tv_nsec;
#else
  clock_gettime(CLOCK_REALTIME, ts);
#endif

}


int main(int argc, char **argv) {

  struct timespec ts;
  current_utc_time(&ts);

  printf("s:  %lu\n", ts.tv_sec);
  printf("ns: %lu\n", ts.tv_nsec);
  return 0;

}
```

**Comment by shamis** [comment]
fixed in #151
